index.js

```diff
@@ -195,24 +195,29 @@ function PgSimplifyInflectorPlugin(
           ? proc.tags.fieldName + ListSuffix(proc)
           : this.camelCase(pseudoColumnName + listSuffix(proc));
       },
-      singleRelationByKeys(detailedKeys, table, _foreignTable, constraint) {
+      singleRelationByKeys(detailedKeys, table, foreignTable, constraint) {
         if (constraint.tags.fieldName) {
           return constraint.tags.fieldName;
         }
         const baseName = this.getBaseNameFromKeys(detailedKeys);
         if (baseName) {
           return this.camelCase(baseName);
         }
-        if (this.baseNameMatches(baseName, table.name)) {
+        if (this._isPrimaryKey(detailedKeys, foreignTable)) {
```

Comment:
Please can you restore the `baseNameMatches` call, or explain why it has been removed?

Reply:
If I'm not wrong, it's not possible for this condition to be true, because the previous construct is

```
if (baseName) {
          return this.camelCase(baseName);
}
```

So `baseName` must be falsey here. From the implementation of `baseNameMatches`, it shouldn't be possible for `table.name` to match something falsey.

Reply:
The refactoring of `oppositeBaseName` and friends follows the same logic.

Reply:
Good catch :+1: